src/Microsoft.AspNetCore.Http.Connections/HttpConnectionManager.cs

```diff
@@ -28,34 +28,24 @@ public partial class HttpConnectionManager
 
         private readonly ConcurrentDictionary<string, (HttpConnectionContext Connection, ValueStopwatch Timer)> _connections =
             new ConcurrentDictionary<string, (HttpConnectionContext Connection, ValueStopwatch Timer)>(StringComparer.Ordinal);
-        private Timer _timer;
+        private TimerAwaitable _timer;
         private readonly ILogger<HttpConnectionManager> _logger;
         private readonly ILogger<HttpConnectionContext> _connectionLogger;
-        private readonly object _executionLock = new object();
-        private bool _disposed;
 
         public HttpConnectionManager(ILoggerFactory loggerFactory, IApplicationLifetime appLifetime)
         {
             _logger = loggerFactory.CreateLogger<HttpConnectionManager>();
             _connectionLogger = loggerFactory.CreateLogger<HttpConnectionContext>();
             appLifetime.ApplicationStarted.Register(() => Start());
             appLifetime.ApplicationStopping.Register(() => CloseConnections());
+            _timer = new TimerAwaitable(_heartbeatTickRate, _heartbeatTickRate);
```

Comment:
Since we're going to be `await`ing this, maybe give it a name that makes sense for that like `_nextHeartbeat`. So the code is `await _nextHeartbeat`